### ISSUE TITLE
`latexml --VERSION` crashes GitHub action

### ISSUE THREAD

**Issue by FelixBenning** [brucemiller/LaTeXML#1845]
This is a weird bug. Using the github action environment (`ubuntu-latest`), the following code
```yaml
      - name: Install Dependencies
        run: |
            sudo apt-get update && sudo apt-get upgrade -y
            sudo apt-get install latexml
            latexml --VERSION
```

resulted in

```shell
 latexml (LaTeXML version 0.8.4)
Error: Process completed with exit code 1.
```

I tried using `latexml --VERSION` locally, but got a zero exit code 🤷. After removing the query for the version my pipeline worked again.

This is probably not that important

**Comment by dginev** [comment]
I guess we can give it an exit code of `0`, sounds like a reasonable expectation.

**Comment by dginev** [comment]
> Oh - interesting. So it apparently works with the wrong casing locally that is weird.

In case you didn't know - some operating systems have case-insensitive filenames, MacOS being the usual example. But most standard linux OSes are case-sensitive, and I think github CI is using such a system by default.

**Comment by FelixBenning** [comment]
yeah - not used to MacOS...  just started using that at work. That will be the reason

**Comment by dginev** [comment]
```
latexmlc --whatsin=math 'literal:1' \
   --xsltparameter=LATEJS:true \
   --javascript=LaTeXML-maybeMathjax.js \
   --format=html5
```

This works for me locally, javascript tag deposited just before the closing `</body>` tag.
 Maybe you can share the full command that fails to use the flag?

**Comment by FelixBenning** [comment]
sure - I am not very happy with its current state (I could not get the list form of it to work) but here it is

```python
    # make html
    subprocess.run(
        " ".join(
            [
                "latexmlpost",
                f'--dest="{html_path.absolute()}"',
                '--javascript="LaTeXML-maybeMathjax.js"',
                '--javascript="js/collapsing_proofs.js"',
                "--xsltparameter=LATEJS:true", # seems to be broken for github actions
                f'"{xml_path.absolute()}"',
            ]
        ),
        shell=True,
        check=True,
        cwd=destination,
    )
```

it is part of a bunch of command line interface programs I build with [typer](https://typer.tiangolo.com/) to do these management tasks. So I know for sure that I am using the same command locally as in the github action

**Comment by FelixBenning** [comment]
oh maybe I should try to reorder

**Comment by FelixBenning** [comment]
that did not fix it. But I figured out why I could not get the subprocess.run thing to work without the string manipulations. It does not like the `=` option style...

So it looks like this now

```python
    # make html
    subprocess.run(
        [
            "latexmlpost",
            "--dest",
            html_path.absolute(),
            "--xsltparameter",
            "LATEJS:true",  # seems to be broken for github actions
            '--javascript',
            'LaTeXML-maybeMathjax.js',
            '--javascript',
            'js/collapsing_proofs.js',
            xml_path.absolute(),
        ],
        check=True,
        cwd=destination,
    )
```
not sure if that is cleaner 🤣

**Comment by FelixBenning** [comment]
> ```
> latexmlc --whatsin=math 'literal:1' \
>    --xsltparameter=LATEJS:true \
>    --javascript=LaTeXML-maybeMathjax.js \
>    --format=html5
> ```
> 
> This works for me locally, javascript tag deposited just before the closing `</body>` tag. Maybe you can share the full command that fails to use the flag?

what does `--whatsin=math` and `'literal:1'` do?

**Comment by dginev** [comment]
> what does --whatsin=math and 'literal:1' do?

The whatsin/whatsout options are specific to `latexmlc` and allow to convert small fragments without writing the full self-contained latex document around it. The "math" value automatically starts a math mode as well.

Similarly, `literal:` tells latexml the following is to be interpreted as a verbatim string, and not a file name. 

I think the LATEJS option was introduced in latexml 0.8.5, so as long as that is installed, it ought to work... Curious mystery there.

**Comment by FelixBenning** [comment]
Thank you for the explanation. I'll keep that in mind - might come in useful if I want to call it from python...

Well given that `latexml --VERSION` resulted in the output

```shell
 latexml (LaTeXML version 0.8.4)
Error: Process completed with exit code 1.
```

before I axed it due to it killing the github action, I think this solves our mystery. So I guess wait for apt-get to update?

**Comment by dginev** [comment]
@FelixBenning right, you either need to get a more recent OS, or install latexml from github instead of the package manager.

Btw, I have a project where the github actions CI installs latexml from the latest master, here is a link if useful:
https://github.com/dginev/latexml-runner/blob/main/.github/workflows/CI.yml#L10-L15

**Comment by brucemiller** [comment]
Hi all; What's the current status of this issue?

**Comment by tkw1536** [comment]
Shouldn't this issue just be closed then?

**Comment by dginev** [comment]
Tom is right, this has been corrected some time since the 0.8.4 release, which was being used in the original CI report.

Locally, today I see:
```
$ latexml --VERSION && echo 'OK'
latexml (LaTeXML version 0.8.6; revision b2cb75c8)
OK
```

latexml, latexmlpost, latexmlmath and latexmlc all have `exit(0)` when displaying the version in the latest latexml HEAD. Closing here.
